containers/src/Data/Tree.hs

```diff
@@ -207,33 +211,84 @@ instance Foldable Tree where
       where go !z (Node x ts) = foldl' go (f z x) ts
     {-# INLINE foldl' #-}
 
-    foldr1 f = go id
-      where go k (Node x ts) = foldr (\n k' prev -> f prev (go k' n)) k ts x
-    {-# INLINE foldr1 #-}
+    foldr1 = foldrMap1 id
 
-    foldl1 f (Node x ts) = foldl (foldl f) x ts
+    foldl1 = foldlMap1 id
 
     null _ = False
     {-# INLINE null #-}
 
     elem = any . (==)
     {-# INLINABLE elem #-}
 
-    maximum = foldl1' max
+    maximum = foldlMap1' id max
```

Comment:
Why this change?

Reply:
To avoid defining a `foldl1'`. Do you see a problem?

Reply:
No, I just missed that.